click/core.py

```diff
@@ -1706,6 +1706,8 @@ def __init__(self, param_decls, required=None, **attrs):
                 required = False
             else:
                 required = attrs.get('nargs', 1) > 0
+        if attrs.get('autocompletion') is not None:
+            self.autocompletion =  attrs.pop('autocompletion')
```

Comment:
I think this should actually show up under Parameters rather than explicitly Argument.  Presumably, an Option might also benefit from autocompletion.